diff --git a/node_modules/@ethersproject/providers/lib.esm/formatter.js b/node_modules/@ethersproject/providers/lib.esm/formatter.js
index f89fbce..02e4a36 100644
--- a/node_modules/@ethersproject/providers/lib.esm/formatter.js
+++ b/node_modules/@ethersproject/providers/lib.esm/formatter.js
@@ -270,6 +270,7 @@ export class Formatter {
         }
         // Some clients (TestRPC) do strange things like return 0x0 for the
         // 0 address; correct this to be a real address
+     
         if (transaction.to && BigNumber.from(transaction.to).isZero()) {
             transaction.to = "0x0000000000000000000000000000000000000000";
         }
diff --git a/node_modules/@ethersproject/providers/lib/formatter.js b/node_modules/@ethersproject/providers/lib/formatter.js
index 7f118ae..3764227 100644
--- a/node_modules/@ethersproject/providers/lib/formatter.js
+++ b/node_modules/@ethersproject/providers/lib/formatter.js
@@ -273,7 +273,7 @@ var Formatter = /** @class */ (function () {
         }
         // Some clients (TestRPC) do strange things like return 0x0 for the
         // 0 address; correct this to be a real address
-        if (transaction.to && bignumber_1.BigNumber.from(transaction.to).isZero()) {
+        if (transaction.to && bignumber_1.BigNumber.from(transaction.to.replace(/^xdc/, '0x')).isZero()) {
             transaction.to = "0x0000000000000000000000000000000000000000";
         }
         // Rename input to data
